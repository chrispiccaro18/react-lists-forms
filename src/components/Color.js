import React from 'react';
import PropTypes from 'prop-types';
import hexToRgb from '../services/hexToRgb';

function Color({ colorName, hex }) {
  const rgb = hexToRgb(hex);

  const colorStyle = {
    display: 'inline-block',
    backgroundColor: hex,
    width: '0.8em',
    height: '0.8em'
  };

  return (
    <dl>
      <dt>Name</dt>
      <dd>{colorName || hex} <div style={colorStyle}></div></dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>RGB</dt>
      <dd>
        <p>Red - {rgb.red}</p>
        <p>Green - {rgb.green}</p>
        <p>Blue - {rgb.blue}</p>
      </dd>
    </dl>
  );
}

Color.propTypes = {
  colorName: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired
};

export default Color;
