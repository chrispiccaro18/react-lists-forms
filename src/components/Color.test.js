import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders a Color', () => {
    const wrapper = shallow(<Color colorName={'red'} hex={'#ff0000'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
