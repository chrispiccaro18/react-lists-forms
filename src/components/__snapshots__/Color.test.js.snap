// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders a Color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    red
     
    <div
      style={
        Object {
          "backgroundColor": "#ff0000",
          "display": "inline-block",
          "height": "0.8em",
          "width": "0.8em",
        }
      }
    />
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #ff0000
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    <p>
      Red - 
      255
    </p>
    <p>
      Green - 
      0
    </p>
    <p>
      Blue - 
      0
    </p>
  </dd>
</dl>
`;
